src/ripple/protocol/STObject.h

```diff
@@ -401,7 +401,7 @@ class STObject
         if (!cf)
             throw std::runtime_error ("Wrong field type");
 
-        cf->setValue (value);
+        cf->setValue (std::move (value));
```

Comment:
By-value template parameters can never deduce to a reference type.  So yes, this is safe, *unless* the client explicitly specifies `V` to be an lvalue reference type.  The `static_assert` would protect us against that possibility.